packages/components/src/components/assets/Icon/Icon.tsx

```diff
@@ -3,15 +3,51 @@ import styled, { css, useTheme } from 'styled-components';
 import { Ref, forwardRef, SVGAttributes } from 'react';
 import { ReactSVG } from 'react-svg';
 import { ICONS } from './icons';
+import { UIVariant } from '../../../config/types';
+import { CSSColor, Color, Colors } from '@trezor/theme';
+
+export type IconVariant = Extract<
+    UIVariant,
+    'primary' | 'tertiary' | 'info' | 'warning' | 'destructive'
+>;
+
+const variantColorMap: Record<IconVariant, Color> = {
+    primary: 'iconPrimaryDefault',
+    tertiary: 'iconSubdued',
+    info: 'iconAlertBlue',
+    warning: 'iconAlertYellow',
+    destructive: 'iconAlertRed',
+};
+
+type ColorProps = {
+    $variant?: IconVariant;
+    theme: Colors;
+    $color?: string;
+};
+
+const getColorForIconVariant = ({
+    $variant,
+    theme,
+    $color,
+}: ColorProps): CSSColor | 'inherit' | string => {
+    if ($color !== undefined) {
+        return $color;
+    }
+
+    return $variant === undefined ? theme.iconDefault : theme[variantColorMap[$variant]];
+};
 
 export type IconType = keyof typeof ICONS;
 
-const SvgWrapper = styled.div<{
-    $color: WrapperProps['color'];
+type SvgWrapperProps = {
+    $color: WrapperProps['color'] | undefined;
```

Comment:
why not `$color?: WrapperProps['color']`

Reply:
To make it required so you dont forget to pass it.

Reply:
Why should we pass it even though it's undefined?

Reply:
Its a consequence of using this trick

![image](https://github.com/trezor/trezor-suite/assets/152899911/1e190de8-40db-4f80-a8d0-f79f435178d3)